packages/slate-react/src/plugins/before.js

```diff
@@ -44,7 +48,9 @@ function BeforePlugin() {
     // since it provides more useful information about the range being affected
     // and also preserves compatibility with iOS autocorrect, which would be
     // broken if we called `preventDefault()` on React's synthetic event here.
-    if (SUPPORTED_EVENTS.beforeinput) return true
+    // Since native `onbeforeinput` mainly benefits autocorrect and spellcheck
+    // for mobile, on desktop it brings IME issue, limit its scope for now.
+    if (IS_MOBILE && SUPPORTED_EVENTS.beforeinput) return true
```

Comment:
I think restricting this will mean that desktop events won't be prevented and we'll end up with duplicates? I'm not sure.

Reply:
It's more like 'decent downgrading' instead of restricting. On desktop Safari native `beforeinput` is buggy, so we fallback to former logic, using React's `onBeforeInput`, together with `onCompositionStart`, `onKeyDown` and `onCompositonEnd` just like what draft.js did.

Reply:
My understanding is wrong, got it fixed.